pkg/pgmodel/metadata/metadata.go

```diff
@@ -0,0 +1,49 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package metadata
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/timescale/promscale/pkg/pgmodel/common/schema"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+// MetricMetadata returns metadata corresponding to metric_families.
+func MetricMetadata(conn pgxconn.PgxConn, metric string, limit int) (map[string][]model.Metadata, error) {
+	var (
+		rows pgx.Rows
+		err  error
+	)
+	if metric != "" {
+		rows, err = conn.Query(context.Background(), "SELECT * from "+schema.Prom+".get_metric_metadata($1)", metric)
+	} else {
+		rows, err = conn.Query(context.Background(), "SELECT metric_family, type, unit, help from "+schema.Catalog+".metadata ORDER BY last_seen")
+	}
+	if err != nil {
+		return nil, fmt.Errorf("query metric metadata: %w", err)
+	}
+	defer rows.Close()
+	metricFamilies := make(map[string][]model.Metadata)
+	for rows.Next() {
+		if limit != 0 && len(metricFamilies) >= limit {
```

Comment:
this is not right since we may not have all the metadata for the chosen metrics in the map yet.

Reply:
Maybe we can change the all-metrics query to (note the order by):
```
		rows, err = conn.Query(context.Background(), "SELECT metric_family, type, unit, help from "+schema.Catalog+".metadata ORDER BY metric_name, last_seen")
```

That way we know metrics are all batched together. It's unclear how we should order the metrics to return for the limit

Reply:
The order is random (w.r.t. metric) as per what Prometheus does. Prometheus cares that metric a metric should be respected completely before moving on to next metric when there's a limit.
See 
https://github.com/prometheus/prometheus/blob/7bc11dcb06640ce22b4e15eb52b2c065f97cf79a/web/api/v1/api.go#L1125

The query you suggested will batch the metric_names and then I think it will be same on our side as well.